### ISSUE TITLE
Looking for the ability to bind an application to a specific network interface

### ISSUE THREAD

**Issue by pjones11** [tnodir/fort#222]
Generally, I use Defender FIrewall for basic firewall needs but I use UFW in Linux for complex firewall needs; however, I'm in a situation where I need more complex firewall capabilities in Windows and I'm struggling to find a way to handle this particular problem.

I am essentially trying to build a point-to-point VPN connection for a single application between two remote servers.  ONLY this one application should be communicating over this VPN connection which is typically represented by its own network interface.  With UFW I could simply tell it that this application can only communicate on a specific interface and I thought that Defender Firewall USED to be able to do this, but, it looks like it only supports connecting to Private or Public and not binding to an interface.

Is this a capability that either Defender or Fort supports and I just missed it?  Is it possible to add this in the future?

**Comment by barkoder** [comment]
@tnodir Is it possible to implement this feature in fort?

I use openvpn, but I don't want to allow traffic for some apps through the VPN. I want those apps to continue using my "Real IP".

The apps lack an internal IP/Interface bind function.

It would be great if Fort supported binding a process to a given interface(whose IP is dynamically changing).

**Comment by barkoder** [comment]
> Can you use a local proxy, bind the proxy to desired local IP

I've tried https://github.com/kizzx2/docker-openvpn-client-socks and https://github.com/curve25519xsalsa20poly1305/docker-openvpn-socks5 .

They both need Docker-Desktop(Not available on Win7). It doesn't work in docker-toolbox.

> and setup the apps to use the proxy?

Even if I managed to get openvpn running inside docker on Windows 7, I couldn't find any setting that would get `vmnat.exe`(from VMWare Workstation 14) to connect to the proxy.

I can't proxychains vmnat.exe

-----------

Basically, sometimes I run a VM and I want all `vmnat.exe` traffic to go though my "Local Area Connection" interface, and other times I run a different VM and I want all `vmnat.exe` traffic to go through "OpenVPN TAP-Windows6" interface.
